### ISSUE TITLE
SIM300 (yoda-conditions) false positive: `assert X < datetime.timedelta(...)`

### ISSUE THREAD

**Issue by rdaysky** [astral-sh/ruff#13754]
I implemented an algorithm that can only handle certain values of a parameter. To make sure future updates to the code don’t bring it outside the allowed range, I added an assertion. This triggered SIM300:
```
DURATION = datetime.timedelta(hours=1)

assert DURATION < datetime.timedelta(days=1), "algorithm breaks on large durations"
```
Ruff apparently thinks that a call to a class constructor is the thing being tested and the constant is the reference value, while actually it’s the other way round. Not sure how the definition of the rule could be updated so it would see things like this.

**Comment by charliermarsh** [comment]
I think the rule is "correct" in that the all-capitalized `DURATION` is typically used as the convention to indicate a constant. Why stylize it that way?

**Comment by rdaysky** [comment]
It follows the convention to indicate a constant because it is a constant. And the subsequent assertion compares it against another constant.

**Comment by charliermarsh** [comment]
I think I understand what you're doing but it's a fairly unusual test. You're validating that a statically-defined constant is less than some threshold value? I don't know that it makes sense to adjust the rule in favor of this, it seems pretty rare.
